File: richdoc/richdoc-cli/src/lib/templates.ts
/**
 * Resolves available HTML templates under <skill-root>/templates/.
 */

import { existsSync, readdirSync } from "node:fs";
import { dirname, resolve } from "node:path";
import { fileURLToPath } from "node:url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

export const TEMPLATES_DIR = resolve(__dirname, "..", "..", "..", "templates");

export function listTemplates(): string[] {
	if (!existsSync(TEMPLATES_DIR)) return [];
	return readdirSync(TEMPLATES_DIR)
		.filter((f) => f.endsWith(".html"))
		.map((f) => f.replace(/\.html$/, ""))
		.sort();
}

export function templatePath(name: string): string {
	return resolve(TEMPLATES_DIR, `${name}.html`);
}

export function templateExists(name: string): boolean {
	return existsSync(templatePath(name));
}

File: richdoc/richdoc-cli/src/richdoc_cli/templates.py
"""Resolve available HTML templates shipped with the skill."""

from __future__ import annotations

from pathlib import Path

from .paths import TEMPLATES_DIR


def list_templates() -> list[str]:
    if not TEMPLATES_DIR.is_dir():
        return []
    return sorted(p.stem for p in TEMPLATES_DIR.glob("*.html") if p.is_file())


def template_path(name: str) -> Path:
    return TEMPLATES_DIR / f"{name}.html"


def template_exists(name: str) -> bool:
    return template_path(name).is_file()
